Lib/test/test_gdb.py

```diff
@@ -321,7 +321,21 @@ def test_bytes(self):
 
     def test_strings(self):
         'Verify the pretty-printing of unicode strings'
-        encoding = locale.getpreferredencoding()
+        # We cannot simply call locale.getpreferredencoding() here,
+        # as GDB might have been linked against a different version
+        # of Python with a different encoding and coercion policy
+        # with respect to PEP 538 and PEP 540.
+        out, err = run_gdb(
+            '--eval-command',
+            'python import locale; print(locale.getpreferredencoding())')
+
+        if err:
+            raise RuntimeError(
+                'unable to determine the preferred encoding '
+                'of embedded Python in GDB.')
```

Comment:
Would you mind to include err in the error message? You can use an f-string.

You might also raise the error if encoding is an empty string (if err or not encofing: ...)